### ISSUE TITLE
otelgin uses non-standard error field to transport error information

### ISSUE THREAD

**Issue by bripkens** [open-telemetry/opentelemetry-go-contrib#5252]
### Description

otelgin uses a span attribute `gin.errors` to transport information about errors attached to the handlers/middleware. This is 

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/a111e3e0f4a63eae2eee6755e0f04c9d4574968e/instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go#L86-L93

### Expected behavior

I expect otelgin to either:

 1. Use the standard span status description field for error messages – which is only possible when the status code is `error` according to spec. This is debatable, because we don't know whether these attached errors are the _cause_ of the returned status code.
 2. By recording errors on the span. For example, by doing:

```go
for _, err := range c.Errors {
	attrs := make([]attribute.KeyValue, 0, 2)
        // to be replaced with semconv usage https://opentelemetry.io/docs/specs/semconv/attributes-registry/error/
	attrs = append(attrs, attribute.String("error.type", fmt.Sprintf("%d", err.Type)))
	if err.Meta != nil {
		attrs = append(attrs, attribute.String("meta", fmt.Sprintf("%v", err.Meta)))
	}
	
	span.RecordError(err.Err, oteltrace.WithAttributes(attrs...))
}
```

Option 2. seems a preferable option to me. As this would be leveraging standard tracing fields to transport the information. Previously, there would be one error field and with this change this would be split across multiple attributes that will then permit searching/filtering/grouping in o11y solutions.

I would be happy to contribute option 2.

**Comment by adityaAllen** [comment]
Agreed with assessment. Option 2 would let better grouping of errors and search ability
